enhancements/security/openshift-image-policy.md

```diff
@@ -0,0 +1,338 @@
+
+### API Extensions
+
+As part of implementing this policy, we will drop the following Godocs from [ClusterImagePolicy][ClusterImagePolicy-scope-restriction] and [ImagePolicy-scope-restriction]:
+
+```golang
+// Please be aware that the scopes should not be nested under the repositories of OpenShift Container Platform images.
+// If configured, the policies for OpenShift Container Platform repositories will not be in effect.
+```
+
+[The backing implementation][container-runtime-scope-restriction] will also be removed.
+This guard was [initially intended][scope-restriction-motivation] to protect cluster adminstrators from breaking their clusters by configuring policies that blocked critical images.
+And before Red Hat was publishing signatures for `quay.io/openshift-release-dev/ocp-release` releases, that made sense.
+But now that Red Hat is [almost][OTA-1267] publishing Sigstore signatures for those release images, it makes sense to allow policies covering those images.
+And even if a cluster administrator creates a policy that blocks critical image pulls, PodDisriptionBudgets should keep the Kubernetes API server and related core workloads running for long enough for the cluster administrator to use the Kube API to remove or adjust the problematic policy.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+HyperShift clusters run pods [on the management cluster and in the hosted cluster][hypershift-concepts].
+A ClusterImagePolicy applied to the hosted Kubernetes API server will only be distributed to hosted nodes, and none of those nodes are likely to run an image from `quay.io/openshift-release-dev/ocp-release`.
+However, a ClusterImagePolicy applied to the management cluster's Kubernetes API server would cover both the management cluster's cluster-version operator (CVO), and the CVOs associated with hosted clusters (which run as pods in mangement-cluster namespaces).
+That means that it would take an ImagePolicy to be able to distinguish between constraints on the management cluster's CVO and constraints on the hosted components running on the management cluster, which is [out of scope for this enhancement](#use-an-imagepolicy-for-openshift--namespaces).
+
+#### Standalone Clusters
+
+This proposal is aimed at standalone clusters.
+
+#### Single-node Deployments or MicroShift
+
+This proposal is aimed at single-node OpenShift deployments (SNOs), and will function there the same way it does in standalone clusters.
+It is not expected to affect resource consumption, because pod-launches are expected to be a relatively insignificant fraction of overall pod lifetime, and the `quay.io/openshift-release-dev/ocp-release` scope check covered by this proposal is expected to cheaply exclude most pods.
+
+This proposal is not relevant for MicroShift, which does not run a cluster-version operator or machine-config operator, and which handles trust via RPM signing.
+
+### Implementation Details/Notes/Constraints
```

Comment:
> If it exists, we should verify the status of the ClusterImagePolicy to ensure it is fully deployed before creating the CVO pod to scrape manifests.

This sounds nice.  But while there is some [has been applied][1] wording in the (Cluster)ImagePolicy enhancement, there are some wrinkles like "but what if one compute machine isn't letting us roll out the policy there" that probably shouldn't block CVO pod creation.  I was expecting to just race them out on install, and expecting a control-plane drain if/when the policy rolled out (although I guess I don't see drains explicitly mentioned in the (Cluster)ImagePolicy enhancement) or similar disruption to break the cluster if a pod violated policy.  And today installs have no in-cluster verification, so this kind of racing isn't worse than what we have now, even if we happen to lose the race and not get any install-time pod reschedules.

> Otherwise, it seems like we leave some space for updating to quay.io/random/repo:latest without a signature check. Having ImagePolicy to rely on would simplify this.

For now, we still have the OpenPGP process continuing in parallel.  But yes, before we drop OpenPGP entirely, it seems very important to have a working ImagePolicy in `openshift-cluster-version`.  Out of scope for this initial tiptoe enhancement though.

> ClusterVersion should expose status explaining if there is a problem with ClusterImagePolicy...

The ClusterVersion operator already reports manifest-reconciliation issues via `Failing`, although it would be nice if ClusterImagePolicy experts could clarify what "there is a problem with ClusterImagePolicy" should look like in Go.

> ...if the CVO pod for an update is not starting...

More on the existing `ReleaseAccepted` condition [over here][2].

[1]: https://github.com/openshift/enhancements/blob/4668a0825c59739dfafd2ae661c16cf30f540946/enhancements/api-review/add-ClusterImagePolicy-and-ImagePolicy-for-Signature-Verification.md?plain=1#L310-L315
[2]: https://github.com/openshift/enhancements/pull/1633#discussion_r1619849199

Reply:
> If it exists, we should verify the status of the ClusterImagePolicy to ensure it is fully deployed before creating the CVO pod to scrape manifests.

I've pushed ddbd388f -> 1b7f6dbc, adding a *Delay running a cluster-version operator pod until the image policy is in place* section explaining why I think it's ok to skip this alternative.  Let me know if you think I'm missing a relevant pro or con or wrinkle :).